SYCL/GroupAlgorithm/support.h

```diff
@@ -15,11 +15,10 @@ bool isSupportedDevice(device D) {
 
   if (PlatformName.find("OpenCL") != std::string::npos) {
     std::string Version = D.get_info<sycl::info::device::version>();
-    size_t Offset = Version.find("OpenCL");
-    if (Offset == std::string::npos)
-      return false;
-    Version = Version.substr(Offset + 7, 3);
-    if (Version >= std::string("2.0"))
+
+    // Group collectives are mandatory in OpenCL 2.0 but optional in 3.0.
+    Version = Version.substr(7, 3);
+    if (Version >= "2.0" && Version < "3.0")
```

Comment:
@Pennycook @steffenlarsen I had to update these as well, but it looks like these tests were never running on OpenCL.

Since the OpenCL plugin was trimming the version it always returned just the number without `OpenCL` in front, so I believe it would always get caught by the `Offset == std::string::npos` condition and exit.

As I understand it, the group collectives are mandatory starting in OpenCL 2.0 but they're optional again in OpenCL 3.0, and without using interop I don't think we have a way to know if the implementation supports them. But also I *think* these are mandatory in SYCL2020 so I'm not too sure how we should handle it.

So I think this check is slightly better than before, but we may need to revisit this later on, at the very least it should fix the issues showing up in the CI for the CPU OpenCL which is 3.0 and seemingly doesn't work with the group collectives right now.

Reply:
Group algorithms and sub-groups are mandatory in SYCL.

I think we should somehow be checking for these missing features, so we can give a warning/error when somebody tries to use a backend that's missing certain things.  @bashbaug might have some ideas about how to query things in this case.  Using OpenCL interop might be the way to go.